File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/asistencia/ResumenMensualResponse.java
package pe.albrugroup.schedule_service.entity.response.asistencia;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

/**
 * Resumen mensual CERRADO de un empleado: hechos balanceados (sin dinero). Es el artefacto de
 * hand-off al ms de calculo. Solo existe para meses pasados (snapshot); el mes en curso se consulta
 * en vivo por otra ruta. {@code balanceFinal} es el deficit no compensado (<= 0); {@code minutosExtra}
 * y {@code minutosCompensados} van aparte del balance.
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResumenMensualResponse {

    private Long idEmpleado;
    private Integer anio;
    private Integer mes;
    private Instant fechaCierre;

    private Integer diasLaborables;
    private Integer diasPresente;
    private Integer diasTardanza;
    private Integer diasTardanzaCompensable;
    private Integer diasTardanzaJustificada;
    private Integer diasFalta;

    private Integer minutosObjetivo;
    private Integer minutosTrabajados;
    private Integer balanceFinal;
    private Integer minutosExtra;
    private Integer minutosCompensados;
    private Integer cantidadTardanzas;
}

File: frontend/src/app/shared/models/schedule/resumen-mensual-response.ts
/**
 * Resumen mensual CERRADO de un empleado (snapshot inmutable de un mes pasado). Hechos balanceados, sin
 * dinero. `balanceFinal` es el déficit no compensado (<= 0); `minutosExtra` y `minutosCompensados` van
 * aparte del balance. Solo existe para meses pasados; el mes en curso se deriva del detalle diario.
 */
export interface ResumenMensualResponse {
  idEmpleado: number;
  anio: number;
  mes: number;
  fechaCierre: string | null;

  diasLaborables: number;
  diasPresente: number;
  diasTardanza: number;
  diasTardanzaCompensable: number;
  diasTardanzaJustificada: number;
  diasFalta: number;

  minutosObjetivo: number;
  minutosTrabajados: number;
  balanceFinal: number;
  minutosExtra: number;
  minutosCompensados: number;
  cantidadTardanzas: number;
}
